src/main/reaktivity/protocol.idl

```diff
@@ -38,24 +46,28 @@
             octets[length] bytes;
         }
 
-        struct MqttPacket
+        struct MqttPacketFixedHeader
         {
-            uint8 packetType;
+            uint8 typeAndFlags;
             uint8 remainingLength;
-            octets[remainingLength] packet;
         }
 
         struct MqttByte
         {
             uint8 value;
         }
 
-        struct MqttTopic
+        struct MqttSubscriptionTopic
         {
-            string topic;
+            string16 topicFilter;
             uint8 options;
         }
 
+        struct MqttTopic
+        {
+            string16 topicFilter;
```

Comment:
Rename `topicFilter` to `filter` since it is in the context of `MqttTopic` already.